util/filter_bench.cc

```diff
@@ -0,0 +1,403 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#ifndef GFLAGS
+#include <cstdio>
+int main() {
+  fprintf(stderr, "Please install gflags to run rocksdb tools\n");
+  return 1;
+}
+#else
+
+#include <cinttypes>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include "port/port.h"
+#include "port/stack_trace.h"
+#include "rocksdb/filter_policy.h"
+#include "util/gflags_compat.h"
+#include "util/hash.h"
+#include "util/stop_watch.h"
+
+using GFLAGS_NAMESPACE::ParseCommandLineFlags;
+using GFLAGS_NAMESPACE::RegisterFlagValidator;
+using GFLAGS_NAMESPACE::SetUsageMessage;
+
+DEFINE_int64(seed, 0, "Seed for random number generators");
+
+DEFINE_double(working_mem_size_mb, 200,
+              "MB of memory to get up to among all filters");
+
+DEFINE_uint32(average_keys_per_filter, 10000,
+              "Average number of keys per filter");
+
+DEFINE_uint32(key_size, 16, "Number of bytes each key should be");
+
+DEFINE_uint32(batch_size, 8, "Number of keys to group in each batch");
+
+DEFINE_uint32(bits_per_key, 10, "Bits per key setting for filters");
+
+DEFINE_uint32(m_queries, 200, "Millions of queries for each test mode");
+
+DEFINE_uint32(new_reader_every, 0,
+              "Create new reader ~1/n queries (0 = never re-create)");
+
+DEFINE_bool(quick, false, "Run more limited set of tests, fewer queries");
+
+DEFINE_bool(allow_bad_fp_rate, false, "Continue even if FP rate is bad");
+
+void _always_assert_fail(int line, const char *file, const char *expr) {
+  fprintf(stderr, "%s: %d: Assertion %s failed\n", file, line, expr);
+  abort();
+}
+
+#define always_assert(cond) \
```

Comment:
I don't find a rule but macro seems to be all capital in this code base, like `ALWAYS_ASSERT`

Reply:
"assert" excepted ;)